CMakeLists.txt

```diff
@@ -54,19 +54,29 @@ else ()
 endif ()
 
 
+# Even in "Debug" build type, static linking prevents storing some symbol information and makes debugging more difficult.
+# use this flag to avoid `$1 = <incomplete type>` messages
+if (NOT KEEP_SYMBOL_INFO)
+  set (KEEP_SYMBOL_INFO false CACHE BOOL "Even in Debug build type, static linking prevents storing some symbol information and makes debugging more difficult. Use this flag to avoid `$1 = <incomplete type>` messages" FORCE)
```

Comment:
I think `FORCE` is useless here as the set command in `if` condition will never be executed if it is already in `CMakeCache` .